src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs

```diff
@@ -14,26 +14,38 @@ internal class DpgOutputLibrary : OutputLibrary
     {
         private readonly IReadOnlyDictionary<InputEnumType, EnumType> _enums;
         private readonly IReadOnlyDictionary<InputModelType, Func<ModelTypeProvider>> _modelFactories;
+        private readonly bool _isCadlInput;
 
         public TypeFactory TypeFactory { get; }
         public IEnumerable<EnumType> Enums => _enums.Values;
         public IEnumerable<ModelTypeProvider> Models => _modelFactories.Values.Select(f => f());
         public IReadOnlyList<LowLevelClient> RestClients { get; }
         public ClientOptionsTypeProvider ClientOptions { get; }
 
-        public DpgOutputLibrary(IReadOnlyDictionary<InputEnumType, EnumType> enums, IReadOnlyDictionary<InputModelType, Func<ModelTypeProvider>> modelFactories, IReadOnlyList<LowLevelClient> restClients, ClientOptionsTypeProvider clientOptions)
+        public DpgOutputLibrary(IReadOnlyDictionary<InputEnumType, EnumType> enums, IReadOnlyDictionary<InputModelType, Func<ModelTypeProvider>> modelFactories, IReadOnlyList<LowLevelClient> restClients, ClientOptionsTypeProvider clientOptions, bool isCadlInput)
         {
             TypeFactory = new TypeFactory(this);
             _enums = enums;
             _modelFactories = modelFactories;
+            _isCadlInput = isCadlInput;
             RestClients = restClients;
             ClientOptions = clientOptions;
         }
 
         public override CSharpType ResolveEnum(InputEnumType enumType)
-            => _enums.TryGetValue(enumType, out var typeProvider)
-                ? typeProvider.Type
-                : throw new InvalidOperationException($"No {nameof(EnumType)} has been created for `{enumType.Name}` {nameof(InputEnumType)}.");
+        {
+            if (!_isCadlInput)
```

Comment:
With this fix, swagger-based DPG projects may generate parameter initializer that references enum which is not generated.